src/actions/wrap.js

```diff
@@ -9,33 +9,31 @@ let shallowCloneNode = node => ({
 export default ({ getNode, flat }, { mutate, replace, remove, add }) => {
   // wrapInGroup in place should make node key be the new key and put root _inside_ the thing
   // like replace/wrapInGroup, used at root level
-  let wrapInGroupInPlace = async (path, newNode) => {
+  let wrapInGroupInPlace = (path, newNode) => {
     // Clone the root node since we'll be modifying it in place in the tree
     let node = shallowCloneNode(getNode(path))
 
-    // Remove all children (they'll be readded at the end when we add the shallow clone)
-    await Promise.all(_.map(child => remove(child.path), node.children))
+    // Remove all children (they'll be re-added at the end when we add the shallow clone)
+    _.map(child => remove(child.path), node.children)
 
     // Mutate existing root into new root
-    await mutate(path, {
-      ...newNode,
-      path: [newNode.key],
-    })
+    mutate(path, { ...newNode, path: [newNode.key] })
 
     // Replace flat tree references to root
     flat[encode([newNode.key])] = flat[encode(path)]
     delete flat[encode(path)]
 
     // Add original root as a child of the new root
-    await add([newNode.key], node)
+    return add([newNode.key], node)
```

Comment:
Everything else looks good, but since this method no longer awaits the mutate/removes, you can't await this action reliably anymore because we're not returning those promises. To do this right, we should probably collect all the promises in an array then return a `Promise.all` over all over them

Reply:
Ok I'll do that just in case although I don't think this is a problem because `dispatch` calls `triggerUpdate` which is debounced so I think waiting for the last action is the same as waiting for all of them since only one search will be triggered.

When I read the code though I realized we need a way of doing "transactions" in the client for those cases where the debounce is small enough that multiple searches can be run during something like `wrapGroupInPlace`, e.x. when we set the debounce time to 0.

Reply:
Done.

Reply:
That is a good point - this might be unnecessary in this case. More generally, we can't rely on that theory though because we also support partial updates. We can think through that at a later time :)

Reply:
I undid this change because tests were failing. Collecting the promises was fine and they all executed in order (`remove` -> `mutate` -> `add`). However, once the first `await` in `dispatch` was encountered, control flow moved from the `dispatch` returned by `remove` to the `dispatch` returned by `mutate` and the reactors for `mutate` were executing before the ones for `remove`, causing an error.